src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1767,6 +2278,18 @@ ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
         goto err_lru_cleanup;
     }
 
+    md->umr_cq        = NULL;
+    md->umr_qp        = NULL;
+    md->umr_mkey_hash = NULL;
+    ucs_list_head_init(&md->umr_mkey_pool);
+
+    if (md->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI) {
+        status = uct_ib_mlx5_devx_umr_init(md);
```

Comment:
maybe initialize it on first usage?
if not lets move it to a later stage in md init

Reply:
Lazy initialization is an interesting idea IMO.
Maybe we can elaborate it even a bit more. Basically on the exporter side we don't even need to allocate a aliases hash map, and on the importer side we don't really need QP,CQ, mkey pool. So in principle we can init just the exporter part of the first UMR mkey create. And we can init just the importer part on the first alias put into hash map. Let me implement that if you agree

Reply:
done